docker/lb/docker-compose-lb.yaml

```diff
@@ -0,0 +1,201 @@
+  avalon-shell:
+    container_name: avalon-shell
+    image: avalon-shell-dev
+    build:
+      context: ../../
+      dockerfile: ./docker/Dockerfile
+      args:
+        - DISPLAY=${DISPLAY:-}
+        - XAUTHORITY=~/.Xauthority
+        - http_proxy
+        - https_proxy
+        - no_proxy
+    environment:
+      - DISPLAY
+      - http_proxy
+      - https_proxy
+      - no_proxy
+    volumes:
+      # Below volume mappings are required for DISPLAY settings by heart disease GUI client
+      - /tmp/.X11-unix:/tmp/.X11-unix
+      - ~/.Xauthority:/root/.Xauthority
+    command: |
+      bash -c "tail -f /dev/null"
+    stop_signal: SIGKILL
+    depends_on:
+      - avalon-enclave-manager
+      - avalon-listener-1
+      - avalon-listener-2
+      - avalon-listener-3
+
+  avalon-enclave-manager:
+    container_name: avalon-enclave-manager
+    image: avalon-enclave-manager-dev
+    build:
+      context: ../../
+      dockerfile: ./enclave_manager/Dockerfile
+      args:
+        - http_proxy
+        - https_proxy
+        - no_proxy
+    environment:
+      - http_proxy
+      - https_proxy
+      - no_proxy
+    expose:
+      # ZMQ socket port. 
+      - 5555
+    command: |
+      bash -c "
+        enclave_manager --lmdb_url http://avalon-lmdb:9090
+        tail -f /dev/null
+      "
+    depends_on:
+      - avalon-lmdb
+      - avalon-listener-1
+      - avalon-listener-2
+      - avalon-listener-3
+
+  avalon-lmdb:
+    container_name: avalon-lmdb
+    image: avalon-lmdb-dev
+    build:
+      context: ../../
+      dockerfile: ./shared_kv_storage/Dockerfile
+      args:
+        - http_proxy
+        - https_proxy
+        - no_proxy
+    environment:
+      - http_proxy
+      - https_proxy
+      - no_proxy
+    expose:
+      # Port is where lmdb server will listen for http request.
+      # Port should be same as in bind parameter or lmdb_config.toml
+      - 9090
+    command: |
+      bash -c "
+        kv_storage --bind http://avalon-lmdb:9090
+        tail -f /dev/null
+      "
+
+  avalon-listener-1:
+    container_name: avalon-listener-1
+    image: avalon-listener-dev
+    build:
+      context: ../../
+      dockerfile: ./listener/Dockerfile
+      args:
+        - http_proxy
+        - https_proxy
+        - no_proxy
+    environment:
+      - http_proxy
+      - https_proxy
+      - no_proxy
+    expose:
+      - 1947
+      # ZMQ socket port.
+      - 5555
+    command: |
+      bash -c "
+        avalon_listener --bind http://avalon-listener-1:1947 --lmdb_url http://avalon-lmdb:9090
+        tail -f /dev/null
+        "
+    depends_on:
+      - avalon-lmdb
+
+  avalon-listener-2:
+    container_name: avalon-listener-2
+    image: avalon-listener-dev
+    build:
+      context: ../../
+      dockerfile: ./listener/Dockerfile
+      args:
+        - http_proxy
+        - https_proxy
+        - no_proxy
+    environment:
+      - http_proxy
+      - https_proxy
+      - no_proxy
+    expose:
+      - 1947
```

Comment:
Can all listener services use same port to listen to?

Reply:
Yes. Service names are different!

Reply:
Ok.